packages/deployments/contracts/contracts/nomad-xapps/contracts/relayer-fee-router/RelayerFeeMessage.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.11;
+
+// ============ External Imports ============
+import {TypedMemView} from "../../../nomad-core/libs/TypedMemView.sol";
+
+import "../../../../lib/forge-std/src/console.sol";
+
+library RelayerFeeMessage {
+  // ============ Libraries ============
+
+  using TypedMemView for bytes;
+  using TypedMemView for bytes29;
+
+  // ============ Enums ============
+
+  // WARNING: do NOT re-write the numbers / order
+  // of message types in an upgrade;
+  // will cause in-flight messages to be mis-interpreted
+  enum Types {
+    Invalid, // 0
+    ClaimFees // 1
+  }
+
+  // ============ Modifiers ============
+
+  /**
+   * @notice Asserts a message is of type `_t`
+   * @param _view The message
+   * @param _t The expected type
+   */
+  modifier typeAssert(bytes29 _view, Types _t) {
+    _view.assertType(uint40(_t));
+    _;
+  }
+
+  // ============ Formatters ============
+
+  /**
+   * @notice Formats an claim fees message
+   * @param _recipient The address of the relayer
+   * @param _amount The amount of fees to claim
+   * @param _transactionIds A group of transaction ids to claim for fee bumps
+   * @return The formatted message
+   */
+  function formatClaimFees(
+    address _recipient,
+    uint256 _amount,
```

Comment:
I agree. we don't need to pass `amount`.

Reply:
Some thoughts:

- If we don't send the amount and only the group of transaction ids, then it also make sense not to send the relayerFee within the transfer message, we can just keep track of the fees+bumps per transactionID on a mapping in the origin.
With this approach we'll need to allow partial claims, to cover a scenario where there are too many unclaimed transactions for a relayer, iterating through a big number of transactions to calculate the total could end in a DOS scenario for the relayer 

- If we send the amount and don't want to allow partial claims, we can keep track of a unclaimed total + list of transaction per relayer on the remote domain, then when a claim is initialized in Connext we send the unclaimed total to the RelayerFeeRouter plus a list of transaction id for claiming the bumps.

The first approach is simpler but a relayer may end needing to do multiple claims if the number of unclaimed transactions is big, the second adds complexity but would allow to claim the unclaimed total in one transaction + total/partial bumps

Any way we choose we need let the relayer set the list transactions to claim because the unclaimed list of transactions could grow up to the point to make it impossible to claim all at once